src/tests/select-test/select-test.c

```diff
@@ -0,0 +1,539 @@
+/*
+ *  NASA Docket No. GSC-18,370-1, and identified as "Operating System Abstraction Layer"
+ *
+ *  Copyright (c) 2019 United States Government as represented by
+ *  the Administrator of the National Aeronautics and Space Administration.
+ *  All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Filename: select-test.c
+ *
+ * Purpose: This file contains functional tests for "osapi-select"
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "common_types.h"
+#include "osapi.h"
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+osal_id_t     s_task_id;
+osal_id_t     s2_task_id;
+osal_id_t     s_socket_id;
+osal_id_t     s2_socket_id;
+osal_id_t     c_socket_id;
+osal_id_t     c2_socket_id;
```

Comment:
A quick description of the socket uses would help clarify, or maybe a more general outline of what this test does in the file description.